scripts/check-region-info-compatibility.ts

```diff
@@ -0,0 +1,59 @@
+import * as fs from 'fs';
+import * as path from 'path';
+
+/**
+ * Given two copies of the @aws-cdk/region-info database, check that no facts have disappeared between them
+ */
+
+type AwsCdkLib = typeof import('aws-cdk-lib');
+type RegionInfoPackage = AwsCdkLib['region_info'];
+
+function main(oldPackage: string, newPackage: string) {
+  const breaksFile = path.join(__dirname, '..', 'allowed-breaking-changes.txt');
+  const allowedBreaks = new Set(fs.readFileSync(breaksFile, { encoding: 'utf-8' }).split('\n'));
+
+  const oldPkg: RegionInfoPackage = (require(oldPackage) as AwsCdkLib).region_info;
+  const newPkg: RegionInfoPackage = (require(newPackage) as AwsCdkLib).region_info;
+
+  const oldFacts = definedFacts(oldPkg.Fact);
+  const newFacts = definedFacts(newPkg.Fact);
+
+  const disappearedFacts = oldFacts.filter((oldFact) => !newFacts.some((newFact) => factEq(oldFact, newFact)));
+  if (disappearedFacts.length > 0) {
+    console.log('Facts have disappeared from region fact database (add to allowed-breaking-changes.txt to ignore):');
```

Comment:
Shouldn't we log this only if we are sure to exit with code 1?

Reply:
Yessir